### PR TITLE
8276093: Improve naming in closures to iterate over card sets 

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I have reviews for this renaming change that tries to change naming for some remembered set closures to include `Closure` instead of `Iterator` in their names. Additionally I removed two or so `G1` prefixes for nested classes. Hopefully this improves the situation a bit, although it's still a mess of nested closures.

Testing: local compilation, gha

Thanks,
  Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8276093](https://bugs.openjdk.java.net/browse/JDK-8276093): Improve naming in closures to iterate over card sets

### Reviewers
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to 455488f86a409e1eb70f3f0ddec7add9635926ac
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6430/head:pull/6430` \
`$ git checkout pull/6430`

Update a local copy of the PR: \
`$ git checkout pull/6430` \
`$ git pull https://git.openjdk.java.net/jdk pull/6430/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6430`

View PR using the GUI difftool: \
`$ git pr show -t 6430`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6430.diff">https://git.openjdk.java.net/jdk/pull/6430.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1CardSet.cpp [hunk 1]
```diff
@@ -821,21 +821,22 @@ void G1CardSet::iterate_cards_during_transfer(CardSetPtr const card_set, CardVis
   }
 }
 
-void G1CardSet::iterate_containers(G1CardSetPtrIterator* found, bool at_safepoint) {
+void G1CardSet::iterate_containers(CardSetPtrClosure* found, bool at_safepoint) {
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/g1CardSet.cpp:824]: Please change `found` to closure or cl as well.

File: src/hotspot/share/gc/g1/g1CardSet.cpp [hunk 2]
```diff
@@ -821,21 +821,22 @@ void G1CardSet::iterate_cards_during_transfer(CardSetPtr const card_set, CardVis
...
+// Applied to all card (ranges) of the containers.
 template <typename Closure>
-class G1ContainerCards {
+class G1ContainerCardsClosure {
   Closure& _iter;
   uint _region_idx;
 
 public:
-  G1ContainerCards(Closure& iter, uint region_idx) : _iter(iter), _region_idx(region_idx) { }
+  G1ContainerCardsClosure(Closure& iter, uint region_idx) : _iter(iter), _region_idx(region_idx) { }
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/g1CardSet.cpp:839]: Maybe change `_iter` and `iter` to `_closure` and `closure`?

File: src/hotspot/share/gc/g1/g1CardSet.cpp [hunk 3]
```diff
@@ -851,14 +852,14 @@ class G1ContainerCards {
...
-class G1CardSetIterateCardsIterator : public G1CardSet::G1CardSetPtrIterator {
+class G1CardSetContainersClosure : public G1CardSet::CardSetPtrClosure {
   G1CardSet* _card_set;
   Closure& _iter;
 
 public:
 
-  G1CardSetIterateCardsIterator(G1CardSet* card_set,
-                                Closure& iter) :
+  G1CardSetContainersClosure(G1CardSet* card_set,
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/g1CardSet.cpp:861]: Same as above.

File: src/hotspot/share/gc/g1/g1CardSet.cpp [hunk 4]
```diff
@@ -868,8 +869,8 @@ class G1CardSetIterateCardsIterator : public G1CardSet::G1CardSetPtrIterator {
   }
 };
 
-void G1CardSet::iterate_cards(G1CardSetCardIterator& iter) {
-  G1CardSetIterateCardsIterator<G1CardSetCardIterator, G1ContainerCards> cl(this, iter);
+void G1CardSet::iterate_cards(CardClosure& iter) {
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/g1CardSet.cpp:872]: iter -> closure

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 5]
```diff
@@ -355,19 +355,19 @@ class G1CardSet : public CHeapObj<mtGCCardSet> {
...
   void iterate_cards_or_ranges_in_container(CardSetPtr const card_set, CardOrRangeVisitor& found);
 
-  class G1CardSetPtrIterator {
+  class CardSetPtrClosure {
   public:
     virtual void do_cardsetptr(uint region_idx, size_t num_occupied, CardSetPtr card_set) = 0;
   };
 
-  void iterate_containers(G1CardSetPtrIterator* iter, bool safepoint = false);
+  void iterate_containers(CardSetPtrClosure* iter, bool safepoint = false);
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:363]: iter -> closure

File: src/hotspot/share/gc/g1/g1CardSet.hpp [hunk 6]
```diff
@@ -355,19 +355,19 @@ class G1CardSet : public CHeapObj<mtGCCardSet> {
...
 
-  class G1CardSetCardIterator {
+  class CardClosure {
   public:
     virtual void do_card(uint region_idx, uint card_idx) = 0;
   };
 
-  void iterate_cards(G1CardSetCardIterator& iter);
+  void iterate_cards(CardClosure& iter);
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/g1CardSet.hpp:370]: iter -> closure

File: src/hotspot/share/gc/g1/heapRegionRemSet.inline.hpp [hunk 7]
```diff
@@ -87,7 +87,7 @@ class G1HeapRegionRemSetMergeCardIterator : public G1CardSet::G1CardSetPtrIterat
 
 public:
 
-  G1HeapRegionRemSetMergeCardIterator(G1CardSet* card_set,
+  G1HeapRegionRemSetMergeCardClosure(G1CardSet* card_set,
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/heapRegionRemSet.inline.hpp:90]: `(_)iter` -> `(_)closure`

**Review by kstefanj — APPROVED** [review]
Looks good, some additional iter -> closure renames suggested below.

**Review by albertnetymk — APPROVED** [review]

**Comment by tschatzl** [comment]
Thanks @albertnetymk @kstefanj for your reviews
/integrate

**Pull request closed without merging**